cli/src/program.rs

```diff
@@ -1137,7 +1137,7 @@ fn do_process_program_write_and_deploy(
     let (initial_message, write_messages, balance_needed) =
         if let Some(buffer_authority_signer) = buffer_authority_signer {
             let (initial_instructions, balance_needed) = if let Some(account) = rpc_client
-                .get_account_with_commitment(buffer_pubkey, config.commitment)?
+                .get_account_shared_data_with_commitment(buffer_pubkey, config.commitment)?
```

Comment:
With the new name change should this be `get_writeable/readable_account_with_commitment`?

Reply:
get_account_shared_data* returns AccountSharedData vs get_account* returns Account. It isn't returning a trait or boxed ReadableAccount. The only difference is the type of the thing it returns. And, in trying to avoid additional copies or overhead each time we get an account, I was imagining we'd need to duplicate these code paths.
There is some discussion [here](https://github.com/solana-labs/solana/pull/15691#issuecomment-793034139).